log_level, CI, misc

Ra2-IFV opened the pull request:
It works on my machine

Ra2-IFV committed:
ci(build.yaml): Init Github build CI

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV committed:
fix(justfile): Allow running xmake without input

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV committed:
feat: log_level

Fuck I don't learn CPP, I'm pure C

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV committed:
chore: Improve msgbox messages

Set msgbox title in 'const wchar_t* pMsgboxTitle' in vars.h

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV commented:
```text
$ git rebase dev pr_dev
Auto-merging assets/请读我.txt
CONFLICT (add/add): Merge conflict in assets/请读我.txt
Auto-merging justfile
CONFLICT (add/add): Merge conflict in justfile
Auto-merging kpatch.ini
CONFLICT (add/add): Merge conflict in kpatch.ini
Auto-merging launcher/main.cpp
CONFLICT (add/add): Merge conflict in launcher/main.cpp
Auto-merging src/dllmain.cpp
CONFLICT (add/add): Merge conflict in src/dllmain.cpp
Auto-merging src/vars.h
CONFLICT (add/add): Merge conflict in src/vars.h
Auto-merging xmake.lua
CONFLICT (add/add): Merge conflict in xmake.lua
error: could not apply dcb0c4a... bypass secure
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
hint: Disable this message with "git config set advice.mergeConflict false"
```

😭

Ra2-IFV commented:
😎

Ra2-IFV committed:
fix(dllmain.cpp): Prioritize first run message box

Add MB_SYSTEMMODAL because sometimes l4d2 window appears on top of it.

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV committed:
feat: Open the link to issue on error

It's so disgusting. The windows libraries.

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV committed:
ci: Feat and all I wrote is shit

Add positional arguments to `just build`, $1 is xmake build type, debug or release

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV commented:
@kurikomoe I wonder if you will generously add support for changing the heapsize value

kurikomoe commented:
> @kurikomoe I wonder if you will generously add support for changing the heapsize value

What `heapsize`? Did you mean the heapsize in the command line option `-heapsize xxx` ?

Ra2-IFV commented:
Yeah, isn't it comes with a maximum value of 512M?

kurikomoe commented:
> Yeah, isn't it comes with a maximum value of 512M?

Really? My experience suggests it is possible to set beyond 512 MiB,
Specifically, when running complex mod configurations with L4D2Fix, `-heapsize 1G` will crash while `-heapsize 512M` wont.
Is there any reference that this option is limited to 512MiB?

Ra2-IFV commented:
I don't know... ppl say it's limited and you'll crash like the "too many indices" error

Ra2-IFV commented:
> https://www.bilibili.com/video/BV14p1qBwEmB/
> -heapsize 2096999
> 这个想必作用大家都知道，重点是后面的数值
> 我个人电脑最高能填10485759

😨

kurikomoe commented:
There is some interesting data from the Chinese L4D2 community regarding this. The hard limit for this variable appears to be 2,097,151 (2gb -1kb)
Additionally, `mem_info` logs show that values exceeding 524,288 only result in an increased WaveData (audio) buffer, without benefiting other subsystems.
(in detail, Chinese) https://www.bilibili.com/opus/829357867040505939

Ra2-IFV commented:
> 10485759

1*1024*1024 I guess xD
Thank you for sharing this!

kurikomoe commented:
Additionally, the `kurikomoe/L4D2Fix/dev` branch implements a version.dll proxy solution. I have marked this as `pre-release` because I am not sure about its stability.

There are reports from players indicating that the patching process is not deterministic—in some cases, The patch fails to modify the `sharedapidx9.dll` correctly. Given these potential reliability issues, I have kept it in pre-release status for now.

Ra2-IFV commented:
Oh, I only tested `release` because `release` version is released after `pre-release`

Ra2-IFV commented:
Enough for today I guess. I hate windows
Do you mind changing theses lines or I don't know what name to use to `mkdir and copy build to dir` after `just build`
https://github.com/kurikomoe/L4D2Fix/blob/dcb0c4aaba6e8a8ce7c173d2ce8a2edf7745726f/justfile#L19-L22
<img width="128" height="128" alt="IMG_20251218_222445_131" src="https://github.com/user-attachments/assets/35ab9340-9300-474d-84e6-d238b04974ce" />

Ra2-IFV committed:
docs: Add CHANGES.md USAGE.md

Following markdown syntax.

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV committed:
build(justfile): Add install, uninstall ...

Remove positional arugment from `build`.
`build` now uses a variable named `XMAKE_TYPE`, defaulting to release.
Add `install`, uses a variable named `PREFIX`, defaulting to
{{XMAKE_TYPE}}.

Well, I'm not authorized to change the code, appending only.
Very inappropriate implementation. Shouldn't appear here. Makefile style.

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV committed:
ci: Update with justfile, add release build

Trigger release build on push to branch or tag v* .

Signed-off-by: Ryan Keane <the.ra2.ifv@gmail.com>

Ra2-IFV commented:
1. > Enough for today I guess. I hate windows Do you mind changing theses lines or I don't know what name to use to `mkdir and copy build to dir` after `just build`
   > https://github.com/kurikomoe/L4D2Fix/blob/dcb0c4aaba6e8a8ce7c173d2ce8a2edf7745726f/justfile#L19-L22
   
   Okay I'm not touching them. https://github.com/kurikomoe/L4D2Fix/pull/15/commits/8240154eb5f7d0e888dce22b2560b41410605cfe
2. Possible cleanups `assets/请读我.txt` `third/source-engine` `kpatch.ini > assets/kpatch.ini`
3. Possible to create release with CI

Let me know your thoughts because this is your project.

kurikomoe committed:
lints: update justfile

kurikomoe commented:
> 1. > Enough for today I guess. I hate windows Do you mind changing theses lines or I don't know what name to use to `mkdir and copy build to dir` after `just build`
>    > https://github.com/kurikomoe/L4D2Fix/blob/dcb0c4aaba6e8a8ce7c173d2ce8a2edf7745726f/justfile#L19-L22
>    
>    
>    Okay I'm not touching them. [8240154](https://github.com/kurikomoe/L4D2Fix/commit/8240154eb5f7d0e888dce22b2560b41410605cfe)
> 2. Possible cleanups `assets/请读我.txt` `third/source-engine` `kpatch.ini > assets/kpatch.ini`
> 3. Possible to create release with CI
> 
> Let me know your thoughts because this is your project.

First, thank you for your PR. I was waiting for you to mark it as "Ready for Review" before taking a look.

Regarding your questions:
1. I don't think the `justfile` changes are strictly necessary. Developers working on this project should modify/write their local `justfile` as needed, so there is no need to concern about the `TARGET`.
2. To be honest, I don't see much value in creating a GitHub Action for this workflow since this project is not updated frequently. However, I'm fine with including it if you think it helps.

Aside from those minor points, this PR LGTM.

Regarding the release plan:
Replacing `System.debug` with `System.log_level` introduces a breaking change. Therefore, I will likely defer creating a new release until the next major version.

Ra2-IFV commented:
> GitHub Action

I don't want giant Visual Code on my computer 😡
I use notepad++, yes very painful, owing to laziness, 5 errors in 6 lines 
want log_level > don't want VS > setup CI > error oriented programming > 为一碟醋包饺子

> Replacing System.debug with System.log_level introduces a breaking change

Defaulting to info, I tried my best.

I don't want L4N and it's fancy features, I'm feeling so lucky that I found this project that provides essential fixes. thank you so much, excited

And most importantly you are an excellent and interesting person ^^

*\*...added to WATCHING list...\**
<img width="128" height="128" alt="ResizedImage_2025-12-21_01-28-16_2464" src="https://github.com/user-attachments/assets/1ffca346-9e50-49ce-aa26-5a18e7df7995" />
